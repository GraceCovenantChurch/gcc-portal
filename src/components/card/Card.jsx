import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

import { default as MaterialCard } from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import Button from "@material-ui/core/Button";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  card: {
    height: "100%",
    display: "flex",
    flexDirection: "column",
  },
  cardMedia: {
    paddingTop: "56.25%", // 16:9
  },
  cardContent: {
    flexGrow: 1,
  },
}));

const Card = (props) => {
  const classes = useStyles();

  return (
    <React.Fragment>
      <MaterialCard className={classes.card}>
        <CardMedia
          className={classes.cardMedia}
          image={props.image}
          title={props.heading}
        />
        <CardContent className={classes.cardContent}>
          <Typography gutterBottom variant="h5" component="h2">
            {props.heading}
          </Typography>
          <Typography>{props.description}</Typography>
        </CardContent>
        <CardActions>
          <Button size="small" color="primary" component={Link} to="/dashboard">
            Click Here
          </Button>
        </CardActions>
      </MaterialCard>
    </React.Fragment>
  );
};

Card.propTypes = {
  heading: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Card;
