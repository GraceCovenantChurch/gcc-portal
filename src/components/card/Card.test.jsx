import React from "react";
import { shallow } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import Card from "./Card";
import renderer from "react-test-renderer";

describe("Card component", () => {
  let component, tree, wrapper;

  beforeEach(() => {
    component = renderer.create(
      <BrowserRouter>
        <Card
          heading="randomheading"
          description="You have a recording of an old sermon? Please upload it here!"
          image="https://www.volgacrc.org/editoruploads/images/FaithWebsites/church_mid_img1.jpg"
        />
      </BrowserRouter>
    );
    tree = component.toJSON();

    wrapper = shallow(
      <Card
        heading="randomheading"
        description="You have a recording of an old sermon? Please upload it here!"
        image="https://www.volgacrc.org/editoruploads/images/FaithWebsites/church_mid_img1.jpg"
      />
    );
  });

  it("UNIT: should be defined", () => {
    expect(Card).toBeDefined();
  });

  it("SNAPSHOT: should render correctly", () => {
    expect(tree).toMatchSnapshot();
  });
});
