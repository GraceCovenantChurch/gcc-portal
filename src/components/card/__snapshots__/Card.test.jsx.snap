// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component SNAPSHOT: should render correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardMedia-root makeStyles-cardMedia-2"
    style={
      Object {
        "backgroundImage": "url(\\"https://www.volgacrc.org/editoruploads/images/FaithWebsites/church_mid_img1.jpg\\")",
      }
    }
    title="randomheading"
  />
  <div
    className="MuiCardContent-root makeStyles-cardContent-3"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      randomheading
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      You have a recording of an old sermon? Please upload it here!
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
      href="/dashboard"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Click Here
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
